lib/utils.js

```diff
@@ -50,7 +50,7 @@ async function resolveExecutablePath (cmd) {
       }
     }
   } catch (err) {
-    log.warn(err);
+    log.debug(err);
```

Comment:
how about this one?

Reply:
Error like `Error: Command 'which carthage' exited with code 1` is printed if we remain `warn` here. In the case, the error is expected. We show a message like `✖ xxx was NOT found!` then.

```
info AppiumDoctor Appium Doctor v.1.6.0
info AppiumDoctor ### Diagnostic starting ###
info AppiumDoctor  ✔ The Node.js binary was found at: /Users/kazuaki/.nodebrew/current/bin/node
info AppiumDoctor  ✔ Node version is 10.11.0
info AppiumDoctor  ✔ Xcode is installed at: /Applications/Xcode.app/Contents/Developer
info AppiumDoctor  ✔ Xcode Command Line Tools are installed in: /Applications/Xcode.app/Contents/Developer
info AppiumDoctor  ✔ DevToolsSecurity is enabled.
info AppiumDoctor  ✔ The Authorization DB is set up properly.
WARN AppiumDoctor Error: Command 'which carthage' exited with code 1
WARN AppiumDoctor     at ChildProcess.proc.on.code (/Users/kazuaki/GitHub/appium-doctor/node_modules/teen_process/lib/exec.js:94:19)
WARN AppiumDoctor     at ChildProcess.emit (events.js:182:13)
WARN AppiumDoctor     at maybeClose (internal/child_process.js:962:16)
WARN AppiumDoctor     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
WARN AppiumDoctor  ✖ Carthage was NOT found!
info AppiumDoctor  ✔ HOME is set to: /Users/kazuaki
info AppiumDoctor  ✔ ANDROID_HOME is set to: /Users/kazuaki/Library/Android/sdk
info AppiumDoctor  ✔ JAVA_HOME is set to: /Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home
info AppiumDoctor  ✔ adb exists at: /Users/kazuaki/Library/Android/sdk/platform-tools/adb
info AppiumDoctor  ✔ android exists at: /Users/kazuaki/Library/Android/sdk/tools/android
info AppiumDoctor  ✔ emulator exists at: /Users/kazuaki/Library/Android/sdk/tools/emulator
info AppiumDoctor  ✔ Bin directory of $JAVA_HOME is set
info AppiumDoctor ### Diagnostic completed, one fix needed. ###
```

I change to show `.warn` if the error code is not `1`. (I'll check windows's case later)
Then, if the raised error is expected (no commands found), we show only `✖ Carthage was NOT found!`.